dali/kernels/erase/erase_gpu.h

```diff
@@ -0,0 +1,526 @@
+    coordinate[d] = idx;
+    int64_t offset = offset_base + idx * sample.sample_stride[d];
+    for (int c = region.lo[dC]; c < boundary_c; c++) {
+      coordinate[dC] = c;
+      Worker::template copy_or_erase<channel_dim>(sample, erase_regions, coordinate, offset + c,
+                                                  fill_values);
+    }
+  }
+}
+
+
+struct do_copy {
+  template <int channel_dim, typename T, int ndim>
+  __device__ static void copy_or_erase(erase_sample_desc<T, ndim> sample,
+                                       span<ibox<ndim>> erase_regions, ivec<ndim> coordinate,
+                                       int64_t offset, span<T> fill_values) {
+    (void)erase_regions;
+    (void)coordinate;
+    (void)fill_values;
+    sample.out[offset] = sample.in[offset];
+  }
+};
+
+struct do_erase {
+  template <int channel_dim, typename T, int ndim>
+  __device__ static void copy_or_erase(erase_sample_desc<T, ndim> sample,
+                                       span<ibox<ndim>> erase_regions, ivec<ndim> coordinate,
+                                       int64_t offset, span<T> fill_values) {
+    (void)erase_regions;
+    if (channel_dim == -1) {
+      sample.out[offset] = fill_values[0];
+    } else {
+      sample.out[offset] = fill_values[coordinate[channel_dim]];
+    }
+  }
+};
+
+struct do_copy_or_erase {
+  template <int channel_dim, typename T, int ndim>
+  __device__ static void copy_or_erase(erase_sample_desc<T, ndim> sample,
+                                       span<ibox<ndim>> erase_regions, ivec<ndim> coordinate,
+                                       int64_t offset, span<T> fill_values) {
+    auto fill_value = channel_dim == -1 ? fill_values[0] : fill_values[coordinate[channel_dim]];
+    copy_or_erase<channel_dim>(sample, erase_regions, coordinate, offset, fill_value);
+  }
+
+  template <int channel_dim, typename T, int ndim>
+  __device__ static void copy_or_erase(erase_sample_desc<T, ndim> sample,
+                                       span<ibox<ndim>> erase_regions, ivec<ndim> coordinate,
+                                       int64_t offset, T fill_value) {
+    bool copy = true;
+    for (auto &region : erase_regions) {
+      if (region.contains(coordinate)) {
+        copy = false;
+      }
+    }
+    if (copy) {
+      sample.out[offset] = sample.in[offset];
+    } else {
+      sample.out[offset] = fill_value;
+    }
+  }
+};
+
+template <int channel_dim = -1, typename T, int ndim = 2>
+__global__ void erase_gpu_impl(erase_sample_desc<T, ndim> *samples, ivec<ndim> region_dims,
+                               int erase_regions_count, span<T> fill_values) {
+  const auto region_idx = blockIdx.x;
+  const auto sample_idx = blockIdx.y;
+
+  const auto &sample = samples[sample_idx];
+  ivec<ndim> sample_dims = sample.sample_shape;
+
+  auto region_box = get_region(region_idx, region_dims, sample_dims);
+
+  const auto *erase_regions_ptrs = sample.erase_regions;
+  constexpr int max_regions = 1024;
```

Comment:
This could be calculated as, say, `40kB / sizeof(ibox<ndim>)`

Reply:
Done.
I wonder how we can do an early check if we won't go over that. It would depend on how are the regions spread out.